packages/block-library/src/navigation/edit/index.js

```diff
@@ -73,6 +73,29 @@ import AccessibleDescription from './accessible-description';
 import AccessibleMenuDescription from './accessible-menu-description';
 import { unlock } from '../../lock-unlock';
 
+function useResponsiveMenu( navRef ) {
+	const [ isResponsiveMenuOpen, setResponsiveMenuVisibility ] =
+		useState( false );
+
+	useEffect( () => {
+		if ( ! navRef.current ) {
+			return;
+		}
+
+		const htmlElement = navRef.current.closest( 'html' );
+
+		// Add a `has-modal-open` class to the <html> when the responsive
+		// menu is open. This reproduces the same behavior of the frontend.
+		if ( isResponsiveMenuOpen ) {
+			htmlElement.classList.add( 'has-modal-open' );
+		} else {
+			htmlElement.classList.remove( 'has-modal-open' );
+		}
+	}, [ navRef, isResponsiveMenuOpen ] );
```

Comment:
Should there be a cleanup function in this `useEffect` to be safe? (Returning a function that removes the class if the component is unmounted). Otherwise if someone deletes the Navigation block while the modal is open, the class persists:

https://github.com/user-attachments/assets/072e3df5-70f8-459f-bdfe-f2646ff28373

Reply:
Good call. +1 for cleanup.

Reply:
I was literally about to type this and you already spotted. Yes we should ensure we clean up here 👍

Reply:
Good point! Thank you for the suggestion folks! Added here: https://github.com/WordPress/gutenberg/pull/66093/commits/0850938415896b525b44fb9c7eaf0b5dcad7d7ca